core/self_list.h

```diff
@@ -103,7 +103,7 @@ class SelfList {
 		_FORCE_INLINE_ const SelfList<T> *first() const { return _first; }
 
 		_FORCE_INLINE_ List() {}
-		_FORCE_INLINE_ ~List() { ERR_FAIL_COND(_first != nullptr); }
+		_FORCE_INLINE_ ~List() {}
```

Comment:
This just fixing crash, but still there is somewhere a memory leak which needs to be fixed.
I don't think that is good idea to remove memory leak check in every single element which use list because this leak happens only on iOS

Reply:
Yeah, I think a better workaround would be to use `printf` or similar instead of `ERR_FAIL_COND` to notify of the leak without relying on `OS::print_error` (which crashes probably because `OS` has already been de-initialized at this point).

Reply:
Ok, I'll do it with OS.

Reply:
What I meant is that it should be done **without** `OS`.

Reply:
Isn't it ok to protect OK with `if (!_logger) return;`? I'll look into the deinitialization order then.

Reply:
@akien-mga I finally got what you say about `printf`, sorry, its difficult morning.

Reply:
Yeah that's an option, `OS::print_error` could be changed to:
```
if (unlikely(!_logger)) {
	printf(...);
} else {
	_logger->log_error(p_function, p_file, p_line, p_code, p_rationale, p_type);
}
```

Not sure if the added check would have a significant performance impact.

Reply:
I found a source of the problem, done without touching SelfList or OS.